### PR TITLE
docs: add caveats for commonjs modules transform

### PR DESCRIPTION
**Pull request by JLHwung**
Closes https://github.com/babel/babel/issues/11127

### CODE REVIEW

**Comment by SimenB** [comment]
This doesn't fully cover it though - I'd be specific about `URL`. Both node and the browser (at least Chrome) supports it, but transpilation breaks it. Linking Node docs doesn't really tell the whole picture.

**Comment by nicolo-ribaudo** [comment]
We should mention that Babel only transforms the syntax of import statements/expressions and that it doesn't know about the resolution algorithms used by the different platforms (Node.js, in this case).

**Comment by SimenB** [comment]
That approach seems better, yeah 👍

**Comment by JLHwung** [comment]
@nicolo-ribaudo We don't have a dedicated plugin page for `proposal-dynamic-import`: https://babeljs.io/docs/en/babel-plugin-proposal-dynamic-import I suggest we create it in another PR if we are good with the phrasing.

File: docs/plugin-transform-modules-amd.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ title: @babel/plugin-transform-modules-amd
 sidebar_label: transform-modules-amd
 ---
 
+This plugin transforms ECMAScript modules to [AMD](https://github.com/amdjs/amdjs-api/blob/master/AMD.md). The _syntax_ of Import/Export Statement (`import "./mod.js"`) and Import Expression (`import('./mod.js')`) are transformed to their AMD representation. Note that Babel is unaware of different resolution algorithms between implementations of ECMAScript modules and AMD.
```
Comment (existentialism) [inline docs/plugin-transform-modules-amd.md]: Maybe:

```suggestion
This plugin transforms ECMAScript modules to [AMD](https://github.com/amdjs/amdjs-api/blob/master/AMD.md). Note that only the _syntax_ of import/export statements (`import "./mod.js"`) and import expressions (`import('./mod.js')`) is transformed, as Babel is unaware of the different resolution algorithms between implementations of ECMAScript modules and AMD.
```

?

File: docs/plugin-transform-modules-umd.md [hunk 2]
```diff
@@ -4,6 +4,10 @@ title: @babel/plugin-transform-modules-umd
 sidebar_label: transform-modules-umd
 ---
 
+This plugin transforms ES2015 modules to [UMD](https://github.com/umdjs/umd). Note that only the _syntax_ of import/export statements (`import "./mod.js"`) is transformed, as Babel is unaware of different resolution algorithms between implementations of ES2015 modules and UMD.
+
+:warning This plugin does not support dynamic import (`import('./lazy.js')`).
```
Comment (nicolo-ribaudo) [inline docs/plugin-transform-modules-umd.md]: Nit: the warning sign isn't properly displayed, I think that you should directly paste the emoji.

**Comment by nicolo-ribaudo** [comment]
@JLHwung Docusaurus doesn't support GitHub emojis, you need to copy-paste the warning character :stuck_out_tongue: 
https://deploy-preview-2173--babel.netlify.com/docs/en/babel-plugin-transform-modules-umd

**Pull request merged**